src/api/tasks/Tasks.controller.js

```diff
@@ -0,0 +1,372 @@
+
+  let localCouncil = await getCouncilFromConfigDb(configDb, registration);
+  if (isEmpty(localCouncil)) {
+    let message = `Could not find local council with ID ${fsaId}`;
+    logEmitter.emit(ERROR, message);
+    throw message;
+  }
+
+  let configVersion = registration.registration_data_version
+    ? registration.registration_data_version
+    : "1.6.0";
+  let config = await getConfig(configDb, configVersion);
+
+  if (isEmpty(config)) {
+    let message = `Could not find config ${fsaId} version : ${configVersion}`;
+    logEmitter.emit(ERROR, message);
+    throw message;
+  }
+
+  //this method is in dire need of refactoring...
+  let lcContactConfig = await getLcContactConfig(
+    localCouncil.local_council_url
+  );
+  if (isEmpty(lcContactConfig)) {
+    let message = `Could not find lcContactConfig ${fsaId} ${
+      localCouncil.local_council_url
+    }`;
+    logEmitter.emit(ERROR, message);
+    throw message;
+  }
+
+  await sendNotifications(fsaId, lcContactConfig, registration, config);
+};
+
+const multiSendRegistrationToTascomi = async configDb => async registration => {
+  let fsaId = registration["fsa-rn"];
+
+  let localCouncil = await getCouncilFromConfigDb(configDb, registration);
+  if (isEmpty(localCouncil)) {
+    let message = `Could not find local council with ID ${fsaId}`;
+    logEmitter.emit(ERROR, message);
+    throw message;
+  }
+
+  // DO LOOK UP
+  if (localCouncil.auth) {
+    try {
+      //GOT AUTH DATA
+      await sendTascomiRegistration(registration, localCouncil);
+
+      //SUCCESS
+      await updateStatusInCache(fsaId, "tascomi", TASCOMI_SUCCESS);
+    } catch (e) {
+      //FAIL
+      logEmitter.emit(
+        ERROR,
+        `Could not push to tascomi for ${fsaId} and local council ${
+          localCouncil._id
+        }. Error: "${e.message}"`
+      );
+      await updateStatusInCache(fsaId, "tascomi", TASCOMI_FAIL);
+    }
+  } else {
+    //NOT APPLICABLE - nothing to update
+    await updateStatusInCache(fsaId, "tascomi", TASCOMI_SKIPPING);
+  }
+};
+
+const multiSaveRegistrationsToTempStore = async configDb => async registration => {
+  let fsaId = registration["fsa-rn"];
+
+  //GET LOCAL COUNCIL
+  let localCouncil = await getCouncilFromConfigDb(configDb, registration);
+  if (isEmpty(localCouncil)) {
+    let message = `Could not find local council with ID ${fsaId}`;
+    logEmitter.emit(ERROR, message);
+    throw message;
+  }
+
+  await saveRegistration(registration, fsaId, localCouncil.local_council_url);
+};
+
+const getConfig = async (client, configVersion) => {
+  let configCollection = await ConfigVersionCollection(client);
+  return await configCollection.findOne({ _id: configVersion });
+};
+
+const getRegistration = async (client, fsaId) => {
+  const cachedRegistrations = await CachedRegistrationsCollection(client);
+  return await findOneById(cachedRegistrations, fsaId);
+};
+
+const getCouncilFromConfigDb = async (client, registration) => {
```

Comment:
I must admit this confuses me and we shuld probably discuss.

Reply:
Oh you mean the HOC/decorator pattern for selecting the collection?

Reply:
as per chat early - clipping currying pattern as i think its old stuff from earlier revisions of this feature.